flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/DefaultCheckpointPlanCalculatorTest.java

```diff
@@ -149,23 +151,114 @@ public void testComputeWithMultipleLevels() throws Exception {
     }
 
     @Test
-    public void testWithTriggeredTasksNotRunning() throws Exception {
+    public void testNotRunningOneOfSourcesTriggeredTasksNotRunning() throws Exception {
+        // given: Execution graph builder with one RUNNING source and NOT RUNNING source.
+        FunctionWithException<JobVertexID, ExecutionGraph, Exception> twoSourcesBuilder =
+                (id) ->
+                        new CheckpointCoordinatorTestingUtils.CheckpointExecutionGraphBuilder()
+                                .addJobVertex(id)
+                                .addJobVertex(new JobVertexID())
+                                .setTransitToRunning(vertex -> !vertex.getJobvertexId().equals(id))
+                                .build();
+
+        // when: Creating the checkpoint plan for aligned and unaligned checkpoints.
+        runWithNotRunTask(twoSourcesBuilder, true);
+        runWithNotRunTask(twoSourcesBuilder, false);
```

Comment:
I don't sure that to test both scenarios(aligned and unaligned) in one test is good idea but I didn't find good example to implement that.